namer/namer.cc

```diff
@@ -131,11 +131,11 @@ class NameInserter {
         return move(localExpr);
     }
 
-    bool addAncestor(core::MutableContext ctx, unique_ptr<ast::ClassDef> &klass, unique_ptr<ast::Expression> &node) {
+    void addAncestor(core::MutableContext ctx, unique_ptr<ast::ClassDef> &klass, unique_ptr<ast::Expression> &node) {
```

Comment:
I **think** by using `cast_tree_const` calls and changing this signature to say `const unique_ptr<ast::Expression> &node` you can guarantee that no one attempts to steal parts of these trees in a future change.

Idk how viral that change would be. Have you tried it?